garage.com/src/main/java/xpathproject.txt

```diff
@@ -0,0 +1,33 @@
+1. Page: /text-box. Write the xpath that will return the ‘Full Name’ textbox using the ID attribute. 
+Hint: This element has the ‘input’ tag.
+	"//input[@id='userName']"
+
+
+2. Page: /buttons. Write the xpath that will return all elements on the page with the ‘button’ tag.
+	"//*[@type='button']"
+
+
+3. Page: /webtables. Write the xpath that will return the div with text of ‘Vega’. Ensure to narrow 
+the scope of your xpath to the ReactTable and ensure that your xpath is limited to returning
+cells only. Cells in this table have the class ‘rt-td’. Hint: You must use “Vega” in your xpath.
+	"//div[@class='rt-td']"
+
+
+4. Page: /radio-button. Write the xpath that will return the parent element for the radio button 
+with id=’impressiveRadio’.
+	"//input[@id='impressiveRadio']//parent::div[1]"
```

Comment:
As a matter of best practice avoid using indexes

//input[@id="impressiveRadio"]/div::parent
or
//input[@id="impressiveRadio"]/div/..

Reply:
Resolved.